python/pyarrow/parquet.py

```diff
@@ -807,6 +807,74 @@ def write_table(table, where, row_group_size=None, version='1.0',
         writer.close()
 
 
+def write_to_dataset(table, root_path, partition_cols=None,
+                     preserve_index=True, **kwargs):
+    """
+    Wrapper around parquet.write_table for writing a Table to
+    Parquet format by partitions.
+    For each combination of partition columns and values,
+    a subdirectories are created in the following
+    manner:
+
+    root_dir/
+    ├── group1=value1
+    │   ├── group2=value1
+    │   │   └── <uuid>.parquet
+    │   └── group2=value2
+    │       └── <uuid>.parquet
+    ...
+    └── group1=valueN
+        ├── group2=value1
+        │   └── <uuid>.parquet
+        ...
+        └── group2=valueN
+            └──<uuid>.parquet
+
+    Parameters
+    ----------
+    table : pyarrow.Table
+    root_path : string,
+        The root directory of the dataset
+    partition_cols : list,
+        Column names by which to partition the dataset
+        Columns are partitioned in the order they are given
+    preserve_index : bool,
+        Parameter for instantiating Table; preserve pandas index or not.
+    **kwargs : dict, kwargs for write_table function.
+    """
+    from pyarrow import (
+        Table,
+        compat
+    )
+
+    if not os.path.isdir(root_path):
+        os.mkdir(root_path)
```

Comment:
This needs to work with other kinds of filesystems (see the `filesystem` argument in ParquetDataset)

Reply:
I'll take a look, thanks!